Index: app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/pojo/Mascotas.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/pojo/Mascotas.java	(date 1480962448000)
+++ app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/pojo/Mascotas.java	(revision )
@@ -7,28 +7,26 @@
  */
 public class Mascotas  {
 
+    private int id;
     private int foto;
-    private int fotolike;
-    private int fotohueso;
     private String nombreMascota;
     private int detallelike;
 
 
 
-    public Mascotas(int foto, int detallelike,int fotohueso) {
+    public Mascotas(int foto, int detallelike) {
         this.foto = foto;
         this.detallelike = detallelike;
-        this.fotohueso = fotohueso;
+
     }
 
 
 
-    public Mascotas(int foto, String nombreMascota, int detallelike, int fotolike, int fotohueso) {
+    public Mascotas(int foto, String nombreMascota, int detallelike) {
         this.foto = foto;
         this.nombreMascota = nombreMascota;
         this.detallelike = detallelike;
-        this.fotohueso = fotohueso;
-        this.fotolike = fotolike;
+
     }
 
     public Mascotas(int foto) {
@@ -36,6 +34,10 @@
 
     }
 
+    public Mascotas() {
+
+    }
+
     public int getFoto() {
         return foto;
     }
@@ -60,19 +62,12 @@
         this.detallelike = detallelike;
     }
 
-    public int getFotolike() {
-        return fotolike;
-    }
 
-    public void setFotolike(int fotolike) {
-        this.fotolike = fotolike;
+    public int getId() {
+        return id;
     }
 
-    public int getFotohueso() {
-        return fotohueso;
-    }
-
-    public void setFotohueso(int fotohueso) {
-        this.fotohueso = fotohueso;
+    public void setId(int id) {
+        this.id = id;
     }
 }
Index: app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/adaptador/MascotaDetalleAdaptador.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/adaptador/MascotaDetalleAdaptador.java	(date 1480962448000)
+++ app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/adaptador/MascotaDetalleAdaptador.java	(revision )
@@ -46,7 +46,6 @@
         final Mascotas mascota = mascotas.get(position);
         mascotaViewHolder.imagenperroCDV.setImageResource(mascota.getFoto());
         mascotaViewHolder.tvReitingDcv.setText(toString().valueOf(mascota.getDetallelike()));
-        mascotaViewHolder.imagenreitingCDV.setImageResource(mascota.getFotohueso());
         mascotaViewHolder.circularImageView.setImageResource(mascota.getFoto());
 
 
Index: app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/adaptador/MascotaAdaptador.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/adaptador/MascotaAdaptador.java	(date 1480962448000)
+++ app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/adaptador/MascotaAdaptador.java	(revision )
@@ -38,10 +38,9 @@
     public void onBindViewHolder(final MascotaViewHolder mascotaViewHolder, int position) {
         final Mascotas mascota = mascotas.get(position);
         mascotaViewHolder.imagendogCV.setImageResource(mascota.getFoto());
-        mascotaViewHolder.imglikecv.setImageResource(mascota.getFotolike());
         mascotaViewHolder.tvnombreVC.setText(mascota.getNombreMascota());
         mascotaViewHolder.likes.setText(toString().valueOf(mascota.getDetallelike()));
-        mascotaViewHolder.huesolikes.setImageResource(mascota.getFotohueso());
+
 
         mascotaViewHolder.imglikecv.setOnClickListener(new View.OnClickListener() {
             @Override
Index: app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/database/ConstructorMascota.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/database/ConstructorMascota.java	(date 1480962448000)
+++ app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/database/ConstructorMascota.java	(revision )
@@ -1,5 +1,6 @@
 package jjimenez.app.com.mascotasrecyclerviewactionview.database;
 
+import android.content.ContentValues;
 import android.content.Context;
 import java.util.ArrayList;
 
@@ -21,17 +22,71 @@
 
     public ArrayList<Mascotas> obtenerMascotas(){
 
-        ArrayList<Mascotas> mascotas = new ArrayList<>();
+       /* ArrayList<Mascotas> mascotas = new ArrayList<>();
 
         mascotas.add(new Mascotas(R.drawable.american_pit_bull_terrier,"American PitBull Terrier", 3,R.drawable.dog_bone_48,R.drawable.dog_bone_48_color));
         mascotas.add(new Mascotas(R.drawable.barbet,"Barte", 6,R.drawable.dog_bone_48,R.drawable.dog_bone_48_color));
         mascotas.add(new Mascotas(R.drawable.boxer,"Boxer", 2,R.drawable.dog_bone_48,R.drawable.dog_bone_48_color));
         mascotas.add(new Mascotas(R.drawable.terrier_tibetano,"Terrier Tibetano", 7,R.drawable.dog_bone_48,R.drawable.dog_bone_48_color));
         mascotas.add(new Mascotas(R.drawable.mastin_espa_ol,"Mastin Español", 0,R.drawable.dog_bone_48,R.drawable.dog_bone_48_color));
+    return  mascotas;*/
 
+        BaseDatosMascota bd = new BaseDatosMascota(context);
+        insertarcincoMascotas(bd);
+        return bd.obtenertodaslasmascotas();
 
 
-    return  mascotas;
+    }
+
+    public void insertarcincoMascotas(BaseDatosMascota db){
+
+        ContentValues contentValues = new ContentValues();
+
+        contentValues.put(ConstanteBaseDatos.TABLE_MASCOTA_FOTO, R.drawable.american_pit_bull_terrier );
+        contentValues.put(ConstanteBaseDatos.TABLE_MASCOTA_NOMBREMASCOTA,"American PitBull Terrier" );
+        contentValues.put(ConstanteBaseDatos.TABLET_LIKE_MASCOTAS,3);
+
+
+        db.insertarMascotas(contentValues);
+
+        contentValues = new ContentValues();
+
+        contentValues.put(ConstanteBaseDatos.TABLE_MASCOTA_FOTO, R.drawable.barbet );
+        contentValues.put(ConstanteBaseDatos.TABLE_MASCOTA_NOMBREMASCOTA,"Barter" );
+        contentValues.put(ConstanteBaseDatos.TABLET_LIKE_MASCOTAS,3);
+
+
+        db.insertarMascotas(contentValues);
+
+        contentValues = new ContentValues();
+
+        contentValues.put(ConstanteBaseDatos.TABLE_MASCOTA_FOTO, R.drawable.boxer );
+        contentValues.put(ConstanteBaseDatos.TABLE_MASCOTA_NOMBREMASCOTA,"Boxer" );
+        contentValues.put(ConstanteBaseDatos.TABLET_LIKE_MASCOTAS,3);
+
+
+        db.insertarMascotas(contentValues);
+
+        contentValues = new ContentValues();
+
+        contentValues.put(ConstanteBaseDatos.TABLE_MASCOTA_FOTO, R.drawable.terrier_tibetano );
+        contentValues.put(ConstanteBaseDatos.TABLE_MASCOTA_NOMBREMASCOTA,"Terrier Tibetano" );
+        contentValues.put(ConstanteBaseDatos.TABLET_LIKE_MASCOTAS,3);
+
+
+        db.insertarMascotas(contentValues);
+
+        contentValues = new ContentValues();
+
+        contentValues.put(ConstanteBaseDatos.TABLE_MASCOTA_FOTO, R.drawable.mastin_espa_ol );
+        contentValues.put(ConstanteBaseDatos.TABLE_MASCOTA_NOMBREMASCOTA,"Mastin Español");
+        contentValues.put(ConstanteBaseDatos.TABLET_LIKE_MASCOTAS,3);
+
+
+        db.insertarMascotas(contentValues);
+
+
+
     }
 
 
Index: app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/database/constanteBaseDatos.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/database/constanteBaseDatos.java	(revision )
+++ app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/database/constanteBaseDatos.java	(revision )
@@ -0,0 +1,25 @@
+package jjimenez.app.com.mascotasrecyclerviewactionview.database;
+
+/**
+ * Created by henry on 09/12/2016.
+ */
+
+public final class ConstanteBaseDatos {
+
+    public static final String DATABASE_NAME                  = "mascotas";
+    public static final int DATABASE_VERSION                  = 1;
+    public static final String TABLE_MASCOTA                  = "mascota";
+    public static final String TABLE_MASCOTA_ID               = "id";
+    public static final String TABLE_MASCOTA_FOTO             = "foto";
+    public static final String TABLE_MASCOTA_NOMBREMASCOTA    = "nombre";
+
+
+
+    public static final String TABLET_LIKE_MASCOTAS            = "mascota_like";
+    public static final String TABLET_LIKE_MASCOTAS_ID         = "id";
+    public  static final String TABLET_LIKE_MASCOTAS_ID_MASCOTAS = "id_mascota";
+    public static final String TABLE_MASCOTA_DETALLE_LIKE      = "numero_like";
+
+
+
+}
Index: app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/constantes/Confi.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/constantes/Confi.java	(date 1480962448000)
+++ app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/constantes/Confi.java	(revision )
@@ -58,11 +58,11 @@
         public void inicializarDatos(){
             mascotas = new ArrayList<Mascotas>();
 
-            mascotas.add(new Mascotas(R.drawable.american_pit_bull_terrier,"American PitBull Terrier", 3,R.drawable.dog_bone_48,R.drawable.dog_bone_48_color));
-            mascotas.add(new Mascotas(R.drawable.barbet,"Barte", 6,R.drawable.dog_bone_48,R.drawable.dog_bone_48_color));
-            mascotas.add(new Mascotas(R.drawable.boxer,"Boxer", 2,R.drawable.dog_bone_48,R.drawable.dog_bone_48_color));
-            mascotas.add(new Mascotas(R.drawable.terrier_tibetano,"Terrier Tibetano", 7,R.drawable.dog_bone_48,R.drawable.dog_bone_48_color));
-            mascotas.add(new Mascotas(R.drawable.mastin_espa_ol,"Mastin Español", 0,R.drawable.dog_bone_48,R.drawable.dog_bone_48_color));
+            mascotas.add(new Mascotas(R.drawable.american_pit_bull_terrier,"American PitBull Terrier", 3));
+            mascotas.add(new Mascotas(R.drawable.barbet,"Barte", 6));
+            mascotas.add(new Mascotas(R.drawable.boxer,"Boxer", 2));
+            mascotas.add(new Mascotas(R.drawable.terrier_tibetano,"Terrier Tibetano", 7));
+            mascotas.add(new Mascotas(R.drawable.mastin_espa_ol,"Mastin Español", 0));
 
 
 
Index: app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/database/BaseDatosMascota.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/database/BaseDatosMascota.java	(revision )
+++ app/src/main/java/jjimenez/app/com/mascotasrecyclerviewactionview/database/BaseDatosMascota.java	(revision )
@@ -0,0 +1,93 @@
+package jjimenez.app.com.mascotasrecyclerviewactionview.database;
+
+import android.content.ContentResolver;
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+
+import java.util.ArrayList;
+
+import jjimenez.app.com.mascotasrecyclerviewactionview.pojo.Mascotas;
+
+/**
+ * Created by henry on 09/12/2016.
+ */
+
+public class BaseDatosMascota extends SQLiteOpenHelper {
+
+    private Context context;
+    public BaseDatosMascota(Context context) {
+        super(context, ConstanteBaseDatos.DATABASE_NAME, null, ConstanteBaseDatos.DATABASE_VERSION);
+        this.context = context;
+    }
+
+    @Override
+    public void onCreate(SQLiteDatabase sqLiteDatabase) {
+
+        String queryCrearTablamascota = "CREATE TABLE" + ConstanteBaseDatos.TABLE_MASCOTA + "(" +
+                ConstanteBaseDatos.TABLE_MASCOTA_ID            + "INTEGER PRIMARY KEY AUTOINCREMENT, " +
+                ConstanteBaseDatos.TABLE_MASCOTA_FOTO          + "INTEGER, " +
+                ConstanteBaseDatos.TABLE_MASCOTA_NOMBREMASCOTA + "TEXT, " +
+                ")";
+
+
+        String queryCrearTablalikeMascotas = "CREATE TABLE" + ConstanteBaseDatos.TABLET_LIKE_MASCOTAS + "(" +
+                ConstanteBaseDatos.TABLET_LIKE_MASCOTAS_ID + "INTEGER PRIMARY KEY AUTOINCREMENT, " +
+                ConstanteBaseDatos.TABLET_LIKE_MASCOTAS_ID_MASCOTAS + "INTEGER, " +
+                ConstanteBaseDatos.TABLE_MASCOTA_DETALLE_LIKE + "INTEGER, " +
+                "FOREIGN KEY (" + ConstanteBaseDatos.TABLET_LIKE_MASCOTAS_ID_MASCOTAS + ") " +
+                "REFERENCE" + ConstanteBaseDatos.TABLE_MASCOTA + "("+ ConstanteBaseDatos.TABLE_MASCOTA_ID +")" +
+                ")";
+
+
+        sqLiteDatabase.execSQL(queryCrearTablamascota);
+        sqLiteDatabase.execSQL(queryCrearTablalikeMascotas);
+
+
+
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int i, int i1) {
+        db.execSQL("DROP TABLE IF EXIST" + ConstanteBaseDatos.TABLE_MASCOTA);
+        db.execSQL("DROP TABLE IF EXIST" + ConstanteBaseDatos.TABLET_LIKE_MASCOTAS);
+        onCreate(db);
+
+
+    }
+
+    public ArrayList<Mascotas> obtenertodaslasmascotas(){
+        ArrayList<Mascotas> mascota = new ArrayList<>();
+
+        String query = "SELECT * FROM " + ConstanteBaseDatos.TABLE_MASCOTA;
+        SQLiteDatabase db = this.getWritableDatabase();
+        Cursor registros = db.rawQuery(query, null);
+
+        while (registros.moveToNext()){
+            Mascotas mascotaActual = new Mascotas();
+            mascotaActual.setId(registros.getInt(0));
+            mascotaActual.setFoto(registros.getInt(1));
+            mascotaActual.setNombreMascota(registros.getString(2));
+            mascotaActual.setDetallelike(registros.getInt(3));
+
+            mascota.add(mascotaActual);
+
+        }
+        db.close();
+
+        return mascota;
+
+    }
+
+    public  void insertarMascotas(ContentValues contentValues){
+
+        SQLiteDatabase db = this.getWritableDatabase();
+        db.insert(ConstanteBaseDatos.TABLE_MASCOTA, null, contentValues );
+        db.close();
+
+
+
+    }
+}
